x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/utils.ts

```diff
@@ -27,6 +32,7 @@ export const transformError = (err: Error & { statusCode?: number }): OutputErro
         statusCode: err.statusCode,
       };
     } else if (err instanceof TypeError) {
+      // TODO should we really count an error as a bad request?
```

Comment:
For the answer to this TODO, there are only 3 spots in which we throw a bad request and it is a very rare event for it to occur but it is possible in a few spots such as pre-packaged rules not working correctly or for some reason a validation that should have never allowed code to pass a particular spot still passes and we throw a TypeError to say, "I need a validation error".

We could have probably just done a custom Error type such as ValidationError and that might have been a bit more clear.

These lines of code really shouldn't be getting touched though fwiw (or we messed up badly somewhere else)

Reply:
I don't think `TypeError` really captures the intention here, since it's such a ubiquitous error in JS already. Since we use `transformError` in almost every DE route, any operations on e.g. `undefined` are going to throw a TypeError and we'll ultimately return a 400 due to the above. This may have been intentional behavior (since operating on an `undefined` certainly _could_ be the result of a bad request), but my concern was that this condition was too broad.

I've already got another PR to fix error-handling on the frontend that changes these explicit TypeErrors to our own `BadRequestError`, so if you're fine with that I'll just delete this TODO for this PR.

Reply:
Yep, I agree with you. It can come out in other bad ways and return 400 instead of 500. That is all true. I am fine with us changing this to be `BadRequestError` instead.

Reply:
29b0b237d3